app/src/main/java/org/fossasia/susi/ai/settings/ChatSettingsFragment.kt

```diff
@@ -122,6 +124,40 @@ class ChatSettingsFragment : PreferenceFragmentCompat(), ISettingsView {
             server.isEnabled = false
         }
 
+        if(!settingsPresenter.getAnonymity()) {
+            resetPassword.isEnabled = true
+            resetPassword.setOnPreferenceClickListener {
+                val builder = AlertDialog.Builder(activity)
+                val resetPasswordView = activity.layoutInflater.inflate(R.layout.alert_reset_password, null)
+                val password = resetPasswordView.findViewById(R.id.password) as TextInputLayout
+                val newPassword = resetPasswordView.findViewById(R.id.newpassword) as TextInputLayout
+                val conPassword = resetPasswordView.findViewById(R.id.confirmpassword) as TextInputLayout
+                builder.setView(resetPasswordView)
+                builder.setTitle(Constant.CHANGE_PASSWORD)
+                        .setCancelable(false)
+                        .setNegativeButton(Constant.CANCEL, null)
+                        .setPositiveButton(getString(R.string.ok), null)
+                val alert = builder.create()
+                alert.show()
+                alert.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener {
+                    if(!CredentialHelper.checkIfEmpty(password, activity) && !CredentialHelper.checkIfEmpty(newPassword, activity) && !CredentialHelper.checkIfEmpty(conPassword, activity)) {
```

Comment:
Whenever button is pressed just call `presenter.method`.  It is duty of presenter to check if password is valid or not.